Dimitrov Task 04

DaimonV15 opened the pull request.

DaimonV15 committed:
Merge remote-tracking branch 'upstream/dev' into dev

DaimonV15 committed:
Task03 Dimitrov

распарсил json, суммммировал баланс по фильтру и вывод в файл.
Красивую табл. не успел %((

DaimonV15 committed:
task04

1. Выравнивание по центру
2. Добавил возможность "очищения таблицы" от двойных границ
3. Перенос текста внутри ячейки
4. Выравнивание по вертикали и горизонтали
5. Постарался сделать тесты %))

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+}
+
+/**
+ * Сложение строк с пробелами
```
не оставляйте мертвый код, он ничего не дает и из него не возможно что-то понять

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+    return tempList
+} */
+
+fun listChunked(text: String, width: Int): List<String> {
```
зачем эта функция? в чем разница:
`listChunked("FFFF", 100) и "FFFF".chunked(100) `

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+
+fun maxWidth(vararg strings: String, width: Int): Int {
+    var tempMaxWidth = width
+    for (element in strings) {
```
используйте `forEach`

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+    return text.chunked(width)
+}
+
+fun maxWidth(vararg strings: String, width: Int): Int {
```
`strings.maxBy { string -> string.maxBy { it.length } ?: width } ?: width`

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+    return tempMaxWidth
+}
+
+fun maxHeight(vararg strings: String, width: Int): Int {
```
аналогично, используйте `maxBy`

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+
+fun String.centralization(width: Int): String {
+    val temp: String
+    var emptySpace = width - this.length
```
можно использовать `coerceAtLeast`
`(width - length).coerceAtLeast(2)` или `minOf(...)`

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+fun buildInTable(line: Int, rows: Int): Array<Array<Int>> {
+    val constructorCase = Array(line) { Array(rows) { 4 } }
+    constructorCase[0][0] = 1
+    for (i in 1 until rows) constructorCase[0][i] = 2
```
```
return Array(line) { i -> Array(rows) { j ->
when {
  i == 0 && j == 0 -> 1
  i == 0 -> 2
  j == 0 -> 3
  else -> 4
}
}
```

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -0,0 +1,80 @@
+    return constructorCase
+}
+
+fun counter(value: Int): MutableList<Int> {
```
`collect(value)` - это оно и есть либо `List(value) { it }`

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/Starter.kt:
```diff
@@ -0,0 +1,37 @@
+        log.info { json }
+
+        val maxWidth = maxWidth(
+            json.take(3)[0].name, json.take(3)[0].gender,
```
Сразу при загрузке заберите только 3 элемента ... это не читабельно

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/Cell.kt:
```diff
@@ -0,0 +1,83 @@
+        if (totalHeight<3) totalHeight = 3
+
+        if (clearOtherInTable) {
+            if (tableHorizontal == 1) {
```
используте здесь when

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/Row.kt:
```diff
@@ -0,0 +1,29 @@
+     */
+
+    fun build(): String {
+        for ((index, it) in cells.withIndex()) it.tableHorizontal = tableBuildProtocol[col[0]][index]
```
`forEachIndexed`

DaimonV15 committed:
Task 05 + change Task 04

В 4ом задании поправил почти все что было помечено на гите
В 5ом задании реализовал функции отрисовки точки, линий, квадрата и круга

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -6,69 +6,42 @@ fun String.newMaxHeight(width: Int): Int {
-    for (element in strings) {
-        val temp = element.split(" ")
-        for (items in temp) if (items.length > tempMaxWidth) tempMaxWidth = items.length
+    strings.forEach {
```
```
strings.maxBy { string -> 
    string.split(" ").maxBy { it.length }?.length ?: -1 
}
```

xthebat commented on dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt:
```diff
@@ -6,69 +6,42 @@ fun String.newMaxHeight(width: Int): Int {
     var tempMaxHeight = 1
-    for (element in strings) {
-        val temp = element.chunked(width)
+    strings.forEach {
```
аналогично есть функции maxBy

xthebat commented on dimitrov/sem05/src/main/kotlin/ru/bmstu/iu4/sem05/common/Canvas.kt:
```diff
@@ -0,0 +1,37 @@
+
+    private fun canvas(): Array<CharArray> {
+        val canvasArray = Array(height*resolution) { CharArray(width*resolution) { ' ' } }
+        for (i in 0 until height*resolution) {
```
что это делает? Если это нарисовать границу, то реализуйте рисование прямоугольника и затем создайте прямоугольник и отрисуйте его на это канвасе и все

xthebat commented on dimitrov/sem05/src/main/kotlin/ru/bmstu/iu4/sem05/impl/Circle.kt:
```diff
@@ -0,0 +1,24 @@
+    override fun draw(plot: Plot) {
+        Point(center.x, center.y).draw(plot)
+
+        Line(Point(center.x, center.y - radius), Point(center.x - radius, center.y)).draw(plot)
```
Нууууу это прям очень очень грубая аппроксимация линиями, скорее всего у всегда будет ромб

xthebat commented on dimitrov/sem05/src/main/kotlin/ru/bmstu/iu4/sem05/impl/Line.kt:
```diff
@@ -0,0 +1,92 @@
+class Line(val src: Point, val dst: Point) : Drawable {
+
+    override fun draw(plot: Plot) {
+        if (src.x.toInt() == dst.x.toInt() && src.y.toInt() < dst.y.toInt())
```
x и у лучше сразу привести к интам в начале функции, читать очень сложно
`(srcY until dstY).forEach { plot[srcX][it] = '*' }`

xthebat reviewed (approved):
В task04 не нашел полных тестов 4/5
По task05 местами сложный код 4/5, если исправить можно 5/5
